File: app/src/main/java/me/lukeforit/spaceofaday/ui/archive/items/EmptyApodViewHolder.java
package me.lukeforit.spaceofaday.ui.archive.items;

import android.support.annotation.NonNull;

import me.lukeforit.spaceofaday.databinding.ListItemApodEmptyBinding;
import me.lukeforit.spaceofaday.ui.base.DataBindingViewHolder;

public class EmptyApodViewHolder extends DataBindingViewHolder<ListItemApodEmptyBinding, EmptyApodItem>{

    public EmptyApodViewHolder(@NonNull ListItemApodEmptyBinding binding) {
        super(binding);
    }

    @Override
    public void bind(@NonNull EmptyApodItem data) {
        binding.setItem(data.getDate());
    }
}

File: app/src/main/java/me/lukeforit/spaceofaday/ui/archive/items/EmptyApodViewHolder.kt
package me.lukeforit.spaceofaday.ui.archive.items

import me.lukeforit.spaceofaday.databinding.ListItemApodEmptyBinding
import me.lukeforit.spaceofaday.ui.base.DataBindingViewHolder

class EmptyApodViewHolder(binding: ListItemApodEmptyBinding) : DataBindingViewHolder<ListItemApodEmptyBinding, EmptyApodItem>(binding) {

    override fun bind(data: EmptyApodItem) {
        binding.item = data.date
        binding.moreIv.setOnClickListener { data.listener.onClick(data.date!!) }
    }
}
